core/store/src/db.rs

```diff
@@ -923,8 +943,14 @@ impl RocksDB {
     }
 
     /// Returns version of the database state on disk.
-    pub fn get_version<P: AsRef<std::path::Path>>(path: P) -> Result<DbVersion, DBError> {
-        let db = RocksDB::new_read_only(path)?;
+    pub fn get_version<P: AsRef<std::path::Path>>(
+        path: P,
+        store_config: &StoreConfig,
+    ) -> Result<DbVersion, DBError> {
+        if !store_config.read_only {
+            warn!("consider opening the database in read only mode to get the version");
+        }
```

Comment:
I think it’s better to just make a copy of the config and update `read_only = true` here rather than making it caller’s burden.